airflow/operators/python.py

```diff
@@ -416,8 +418,16 @@ def __init__(
             **kwargs,
         )
         self.string_args = string_args or []
-        self.use_dill = use_dill
-        self.pickling_library = dill if self.use_dill else pickle
+        if use_dill:
+            warnings.warn(
+                "The 'use_dill' parameter is deprecated and will be removed after 01.10.2024. Please use "
+                "'use_cloudpickle' instead. ",
+                AirflowProviderDeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
In case we decide to deprecate the argument (not sure after reading @potiuk [comment](https://github.com/apache/airflow/pull/38531/files#r1540962108) which makes sense), it would be better to update the default value of `use_dill` to `NOTSET` and check if it is set to `True` or `False` in case it is set explicitly to `False`.

Reply:
Yes, so after Jarek's comment I have removed all the deprecations, so now we can accept both parameters